test/unit/ios-log-specs.js

```diff
@@ -55,4 +55,45 @@ describe('system logs', () => {
 
     await log.stopCapture();
   });
+
+  it('should rotate log buffer', async function () {
+    const maxBufferSize = 10;
+    const sliceSizeLimit = maxBufferSize / 2;
+    sliceSizeLimit.should.be.below(maxBufferSize);
+    const logRecordsCount = maxBufferSize * 2;
+    logRecordsCount.should.be.above(maxBufferSize);
+
+    let log = new IOSLog({sim, showLogs: false});
+    log.maxBufferSize = maxBufferSize;
+    log.logIdxSinceLastRequest.should.be.below(0);
+    await log.startCapture();
+    let recentLogs = await log.getLogs();
+    recentLogs.should.have.lengthOf(0);
+    log.logIdxSinceLastRequest.should.be.below(0);
+
+    for (let i = 1; i <= logRecordsCount; ++i) {
+      await fs.writeFile(tmpSystemLog, `${i}\n`, {flag: 'a'});
+      // This is to make sure the new entry has been captured on slow Travis nodes
+      await B.delay(10);
+      if (i >= sliceSizeLimit && i % sliceSizeLimit === 0) {
+        // on some slow system (e.g., Travis) need a moment
+        let previousRecentLogs = recentLogs;
+        recentLogs = await log.getLogs();
+        if (previousRecentLogs.length && recentLogs.length) {
+          previousRecentLogs[0].message.should.not.be.equal(recentLogs[0].message);
+        }
+        recentLogs.should.have.length.within(1, sliceSizeLimit);
+        log.logIdxSinceLastRequest.should.be.within(0, log.logs.length);
+      }
+      log.logs.should.have.length.at.most(maxBufferSize);
+    }
+    log.logs.should.have.length.within(2, maxBufferSize);
+
+    const firstBufferMessage = parseInt(log.logs[0].message, 10);
+    firstBufferMessage.should.be.within(1, logRecordsCount - log.logs.length);
```

Comment:
It is quite tricky to make it working properly on Trqavis, since log records collection itself is based on log file update callbacks and these callbacks are asynchronous. That is why I need all these delays and range verifications instead of simple equality verification.

BTW, I cannot execute unit test locally for this module. The following error is always shown:

```
...

[08:37:29] 'once' errored after 14 s
[08:37:29] Error in plugin 'run-sequence(unit-test)'
Message:
    Cannot find module 'babel-runtime/regenerator'
Details:
    code: MODULE_NOT_FOUND
Stack:
Error: Cannot find module 'babel-runtime/regenerator'
    at Function.Module._resolveFilename (module.js:325:15)
    at Function.Module._load (module.js:276:25)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/Users/elf/code/appium-ios-driver/build/test/unit/app-util-specs.js:5:27)
    at Module._compile (module.js:397:26)
    at Object.Module._extensions..js (module.js:404:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at /Users/elf/code/appium-ios-driver/node_modules/mocha/lib/mocha.js:220:27
    at Array.forEach (native)
    at Mocha.loadFiles (/Users/elf/code/appium-ios-driver/node_modules/mocha/lib/mocha.js:217:14)
    at Mocha.run (/Users/elf/code/appium-ios-driver/node_modules/mocha/lib/mocha.js:469:10)
    at Domain.<anonymous> (/Users/elf/code/appium-ios-driver/node_modules/gulp-mocha/index.js:56:20)
Nick:appium-ios-driver elf$ npm install -g babel-runtime/regenerator
▀ ╢░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░╟
npm ERR! git clone --template=/Users/elf/.npm/_git-remotes/_templates --mirror git@github.com:babel-runtime/regenerator.git /Users/elf/.npm/_git-remotes/git-github-com-babel-runtime-regenerator-git-7a76b76d: Cloning into bare repository '/Users/elf/.npm/_git-remotes/git-github-com-babel-runtime-regenerator-git-7a76b76d'...
npm ERR! git clone --template=/Users/elf/.npm/_git-remotes/_templates --mirror git@github.com:babel-runtime/regenerator.git /Users/elf/.npm/_git-remotes/git-github-com-babel-runtime-regenerator-git-7a76b76d: ERROR: Repository not found.
npm ERR! git clone --template=/Users/elf/.npm/_git-remotes/_templates --mirror git@github.com:babel-runtime/regenerator.git /Users/elf/.npm/_git-remotes/git-github-com-babel-runtime-regenerator-git-7a76b76d: fatal: Could not read from remote repository.
npm ERR! git clone --template=/Users/elf/.npm/_git-remotes/_templates --mirror git@github.com:babel-runtime/regenerator.git /Users/elf/.npm/_git-remotes/git-github-com-babel-runtime-regenerator-git-7a76b76d: 
npm ERR! git clone --template=/Users/elf/.npm/_git-remotes/_templates --mirror git@github.com:babel-runtime/regenerator.git /Users/elf/.npm/_git-remotes/git-github-com-babel-runtime-regenerator-git-7a76b76d: Please make sure you have the correct access rights
npm ERR! git clone --template=/Users/elf/.npm/_git-remotes/_templates --mirror git@github.com:babel-runtime/regenerator.git /Users/elf/.npm/_git-remotes/git-github-com-babel-runtime-regenerator-git-7a76b76d: and the repository exists.
npm ERR! Darwin 16.4.0
npm ERR! argv "/usr/local/Cellar/node/5.4.1_1/bin/node" "/usr/local/bin/npm" "install" "-g" "babel-runtime/regenerator"
npm ERR! node v5.4.1
npm ERR! npm  v3.5.3
npm ERR! code 128

npm ERR! Command failed: git clone --template=/Users/elf/.npm/_git-remotes/_templates --mirror git@github.com:babel-runtime/regenerator.git /Users/elf/.npm/_git-remotes/git-github-com-babel-runtime-regenerator-git-7a76b76d
npm ERR! Cloning into bare repository '/Users/elf/.npm/_git-remotes/git-github-com-babel-runtime-regenerator-git-7a76b76d'...
npm ERR! ERROR: Repository not found.
npm ERR! fatal: Could not read from remote repository.
npm ERR! 
npm ERR! Please make sure you have the correct access rights
npm ERR! and the repository exists.
npm ERR! 
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/elf/code/appium-ios-driver/npm-debug.log

```